packages/askar/src/wallet/AskarWallet.ts

```diff
@@ -0,0 +1,745 @@
+      this.logger.error(errorMessage, {
+        error,
+        errorMessage: error.message,
+      })
+
+      throw new WalletError(errorMessage, { cause: error })
+    }
+  }
+
+  public async export(exportConfig: WalletExportImportConfig) {
+    // TODO
+    throw new WalletError('AskarWallet Export not yet implemented')
+  }
+
+  public async import(walletConfig: WalletConfig, importConfig: WalletExportImportConfig) {
+    // TODO
+    throw new WalletError('AskarWallet Import not yet implemented')
+  }
+
+  /**
+   * @throws {WalletError} if the wallet is already closed or another error occurs
+   */
+  public async close(): Promise<void> {
+    this.logger.debug(`Closing wallet ${this.walletConfig?.id}`)
+    if (!this._store) {
+      throw new WalletError('Wallet is in invalid state, you are trying to close wallet that has no handle.')
+    }
+
+    try {
+      await this.session.close()
+      await this.store.close()
+      this._session = undefined
+      this._store = undefined
+      this.publicDidInfo = undefined
+    } catch (error) {
+      const errorMessage = `Error closing wallet': ${error.message}`
+      this.logger.error(errorMessage, {
+        error,
+        errorMessage: error.message,
+      })
+
+      throw new WalletError(errorMessage, { cause: error })
+    }
+  }
+
+  public async initPublicDid(didConfig: DidConfig) {
+    // Not implemented, as it does not work with legacy Ledger module
+  }
+
+  /**
+   * Create a key with an optional seed and keyType.
+   * The keypair is also automatically stored in the wallet afterwards
+   *
+   * @param seed string The seed for creating a key
+   * @param keyType KeyType the type of key that should be created
+   *
+   * @returns a Key instance with a publicKeyBase58
+   *
+   * @throws {WalletError} When an unsupported keytype is requested
+   * @throws {WalletError} When the key could not be created
+   */
+  public async createKey({ seed, keyType }: WalletCreateKeyOptions): Promise<Key> {
+    try {
+      if (keyTypeSupportedByAskar(keyType)) {
+        const algorithm = keyAlgFromString(keyType)
+
+        // Create key from seed
+        const key = seed
+          ? AskarKey.fromSeed({ seed: new TextEncoder().encode(seed), algorithm })
+          : AskarKey.generate(algorithm)
+
+        // Store key
+        await this.session.insertKey({ key, name: encodeToBase58(key.publicBytes) })
+        return Key.fromPublicKey(key.publicBytes, keyType)
+      } else {
+        // Check if there is a signing key provider for the specified key type.
+        if (this.signingKeyProviderRegistry.hasProviderForKeyType(keyType)) {
+          const signingKeyProvider = this.signingKeyProviderRegistry.getProviderForKeyType(keyType)
+
+          const keyPair = await signingKeyProvider.createKeyPair({ seed })
+          await this.storeKeyPair(keyPair)
+          return Key.fromPublicKeyBase58(keyPair.publicKeyBase58, keyType)
+        }
```

Comment:
Can askar store keys that it doesn't support? So we still create the key using the SigningKeyProviderRegistry, but we store it using askar?

Reply:
Well, that's what we are doing now, so I guess you mean to store using Askar's specific API to do so (e.g. `session.insertKey `instead of `session.insert`). I don't think it will work because to create a `Key` object it will check `KeyAlgs` enum and it will not be there, so most likely it will throw an `Unsupported` error.